web/src/components/pagination/Pagination.tsx

```diff
@@ -0,0 +1,114 @@
+import { ChevronLeft, ChevronRight } from "lucide-react";
+import Button from "@/refresh-components/buttons/Button";
+
+interface PaginationProps {
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+}
+
+export function Pagination({
+  currentPage,
+  totalPages,
+  onPageChange,
+}: PaginationProps) {
+  // Generate page numbers to display
+  const getPageNumbers = () => {
+    const pages: (number | string)[] = [];
+    const maxPagesToShow = 7;
+
+    if (totalPages <= maxPagesToShow) {
+      // Show all pages if total is small
+      for (let i = 1; i <= totalPages; i++) {
+        pages.push(i);
+      }
+    } else {
+      // Always show first page
+      pages.push(1);
+
+      // Calculate range around current page
+      let startPage = Math.max(2, currentPage - 1);
+      let endPage = Math.min(totalPages - 1, currentPage + 1);
+
+      // Adjust range if we're near the start or end
+      if (currentPage <= 3) {
+        endPage = 5;
+      } else if (currentPage >= totalPages - 2) {
+        startPage = totalPages - 4;
+      }
+
+      // Add ellipsis if needed
+      if (startPage > 2) {
+        pages.push("...");
+      }
+
+      // Add middle pages
+      for (let i = startPage; i <= endPage; i++) {
+        pages.push(i);
+      }
+
+      // Add ellipsis if needed
+      if (endPage < totalPages - 1) {
+        pages.push("...");
+      }
+
+      // Always show last page
+      pages.push(totalPages);
+    }
+
+    return pages;
+  };
+
+  const pageNumbers = getPageNumbers();
+
+  return (
+    <div className="flex items-center justify-center gap-1 mt-4">
+      {/* Previous button */}
+      <Button
+        onClick={() => onPageChange(currentPage - 1)}
+        disabled={currentPage === 1}
+        tertiary
+        className="px-3 py-2"
+      >
+        <ChevronLeft className="h-4 w-4" />
+      </Button>
+
+      {/* Page numbers */}
+      {pageNumbers.map((page, index) => {
+        if (page === "...") {
+          return (
+            <span key={`ellipsis-${index}`} className="px-3 py-2 text-text-500">
+              ...
+            </span>
+          );
+        }
+
+        const pageNum = page as number;
+        const isActive = pageNum === currentPage;
+
+        return (
+          <Button
+            key={pageNum}
+            onClick={() => onPageChange(pageNum)}
+            tertiary={!isActive}
+            className={`px-3 py-2 min-w-[40px] ${
+              isActive ? "bg-background-strong text-inverted" : ""
+            }`}
```

Comment:
Oof, we should definitely remove this. Custom padding should be discouraged, and the text/background colours should not be custom specified either.